src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -0,0 +1,230 @@
+                    else if (elementType is not AggregateType)
+                    {
+                        // Primitives get marshalled as a SingleColumn table.
+                        // Make sure this is consistent with invoker. 
+                        var r2 = new RecordType().Add(TableValue.ValueName, elementType);
+                        return r2.ToTable();
+                    }
+                    else
+                    {
+                        throw new NotImplementedException();                    
+                    }
+
+                case "object":
+                    var obj = new RecordType();
+                    foreach (var kv in schema.Properties)
+                    {
+                        var propName = kv.Key;
+                        var propType = kv.Value.ToFormulaType();
+
+                        obj = obj.Add(propName, propType);
+                    }
+
+                    return obj;
+
+                case null:
+                    // Missing a schema is ok - that signifies void. 
+                    // But if we do have a schema, it must be valid.
+                    throw new InvalidOperationException($"Null/Invalid schema");
+            }
+
+            throw new NotImplementedException($"{schema.Type}");
+        }
+
+        public static HttpMethod ToHttpMethod(this OperationType key)
+        {
+            switch (key)
+            {
+                case OperationType.Get: return HttpMethod.Get;
+                case OperationType.Put: return HttpMethod.Put;
+                case OperationType.Post: return HttpMethod.Post;
+                case OperationType.Delete: return HttpMethod.Delete;
+                case OperationType.Options: return HttpMethod.Options;
+                case OperationType.Head: return HttpMethod.Head;
+                case OperationType.Trace: return HttpMethod.Trace;
+                default:
+                    return new HttpMethod(key.ToString());
+            }
+        }
+
+        public static FormulaType GetReturnType(this OpenApiOperation op)
+        {
+            var responses = op.Responses;
+            if (!responses.TryGetValue("200", out OpenApiResponse response200))
+            {
+                // If no 200, but "default", use that. 
+                if (!responses.TryGetValue("default", out response200))
+                {
+                    throw new NotImplementedException($"Operation must have 200 response ({op.OperationId}).");
+                }
+            }
+
+            if (response200.Content.Count == 0)
+            {
+                // No return type. Void() method. 
+                return FormulaType.Blank;
+            }
+
+            // Responses is a list by content-type. Find "application/json"
+            foreach (var kv3 in response200.Content)
+            {
+                var mediaType = kv3.Key;
+                var response = kv3.Value;
+
+                if (mediaType == "application/json")
```

Comment:
Should we do a case insensitive comparison?
[link](https://stackoverflow.com/questions/7718476/are-http-headers-content-type-c-case-sensitive)